packages/composables/src/composables/useFloating.ts

```diff
@@ -0,0 +1,226 @@
+import { FzFloatingPosition, FzUseFloatingArgs } from '../types'
+import { ref, reactive, onUnmounted } from 'vue'
+import { getHighestAvailableSpacePos } from '../utils'
+
+export const useFloating = (args: FzUseFloatingArgs) => {
+  const safeElementDomRef = ref<HTMLElement | null>(null)
+  const safeContainerDomRef = ref<HTMLElement | null>(null)
+  const safeOpenerDomRef = ref<HTMLElement | null>(null)
+  const rect = ref<DOMRect | null>(null)
+  const float = reactive({
+    position: { x: 0, y: 0 }
+  })
+  const options: IntersectionObserverInit = {
+    root: null,
+    rootMargin: '0px',
+    threshold: 1.0,
+    ...args.element.intersectionOptions
+  }
+
+  const handleIntersect = (
+    entries: IntersectionObserverEntry[],
+    observer: IntersectionObserver
+  ) => {}
+
+  const floatObserver = ref(new IntersectionObserver(handleIntersect, options))
+
+  const setPosition = () => {
+    safeElementDomRef.value =
+      typeof args.element.domRef.value === 'string'
+        ? document.querySelector(args.element.domRef.value)
+        : args.element.domRef.value
+    if (!args.container) {
+      safeContainerDomRef.value = document.body
+    } else {
+      safeContainerDomRef.value =
+        typeof args.container.domRef.value === 'string'
+          ? document.querySelector(args.container.domRef.value)
+          : args.container.domRef.value
+      safeContainerDomRef.value ??= document.body
+    }
+
+    if (!safeElementDomRef.value) {
+      throw new Error('missing reference element for floating behavior')
+    }
+
+    if (args.opener) {
+      safeOpenerDomRef.value =
+        typeof args.opener.domRef.value === 'string'
+          ? document.querySelector(args.opener.domRef.value)
+          : args.opener.domRef.value
+    }
+
+    rect.value = safeElementDomRef.value.getBoundingClientRect()
+    const openerRect = safeOpenerDomRef.value?.getBoundingClientRect()
+    const containerRect = safeContainerDomRef.value.getBoundingClientRect()
+
+    // multiple observer calls on same target do not cause multiple registrations
+    floatObserver.value.observe(safeElementDomRef.value)
+    floatObserver.value.observe(safeContainerDomRef.value)
+
+    let position: FzFloatingPosition = args.position ? args.position.value : 'auto'
+    let translateY = ''
+    let translateX = ''
+
+    if (args.opener && safeOpenerDomRef.value && openerRect) {
+      switch (position) {
```

Comment:
è corretta l'assenza dei break nei vari case in questa switch?

Reply:
si, ammetto che è un po' hacky ma era un attimo un casino cercare di non ripetere della logica sotto, per cui uso questo trick per rientrare dalla finestra nello switch sotto